core/server/worker/src/main/java/alluxio/worker/block/reviewer/AllocationCoordinator.java

```diff
@@ -0,0 +1,52 @@
+package alluxio.worker.block.reviewer;
+
+import alluxio.worker.block.BlockMetadataView;
+import alluxio.worker.block.BlockStoreLocation;
+import alluxio.worker.block.allocator.Allocator;
+import alluxio.worker.block.meta.StorageDirView;
+
+public class AllocationCoordinator {
+  public Reviewer mReviewer;
+  public Allocator mAllocator;
+
+  private static AllocationCoordinator sInstance = null;
+
+  private AllocationCoordinator(BlockMetadataView view) {
+    mAllocator = Allocator.Factory.create(view);
+    mReviewer = Reviewer.Factory.create();
+  }
+
+  public static AllocationCoordinator getInstance(BlockMetadataView view) {
+    if (sInstance == null) {
+      synchronized (AllocationCoordinator.class) {
+        if (sInstance == null) {
+          sInstance = new AllocationCoordinator(view);
+        }
+      }
+    }
+    return sInstance;
+  }
+
+  /**
+   * This is currently only used by tests, where the singleton state needs to be reconfigured.
+   * */
+  public static synchronized void destroyInstance() {
```

Comment:
This is only used by tests where we want to update the reviewer logic or refresh the stateful Allocator.

Reply:
Actually I removed the `synchronized` cuz I don't think it's necessary at the moment.

Reply:
The `TieredBlockStore` is not threadsafe itself.